Ruben-Sten/TeXiFy-IDEA
# Add Tectonic to compiler list

Resolves #888.

#### Additions

* Added Tectonic to compiler list: https://tectonic-typesetting.github.io

[Edit] 
- [x] The Tectonic developer gave me this list of command line arguments, we should support the relevant ones https://github.com/tectonic-typesetting/tectonic/blob/master/src/bin/tectonic.rs#L158

- [x] For relevant compilers, do not auto create a bibtex run config when unnecessary
- [x] For relevant compilers, remove the bibtex run config when switching 
- [x] Show supported output formats per compiler

_Testcase:_

Take a file with bibtex, e.g.
```latex
\documentclass{article}
\begin{document}
    I~\cite{knuth1990}.
    \bibliography{books}
    \bibliographystyle{plain}
\end{document}
```
```
@Book{knuth1990,
    author    = {Knuth, Donald E.},
    title     = {The {\TeX}book },
    year      = {1990},
    isbn      = {0-201-13447-0},
    publisher = {Addison\,\textendash\,Wesley},
}
```

- Delete the run config if it exists.
- Let texify create a new one and run it
- Check the bibliography run config is also run and the result is correct
- Open the run config and change to tectonic (or latexmk, tectonic has internet issues again for me now), see that the bibliographypanel is hidden
- Run it and check that the bib run config is not run (possibly close the open run config windows)


I didn't manage to regenerate the run config when the user changes back to a compiler which does not include bibtex: somehow it doesn't seem to rerender when I unhide elements. Also, changing the value of `bibliographyPanel.configuration` triggers an update which causes the same `applyEditorTo` to be called, causing an infinite loop. And I don't think it's that important.
I tried to rerender by calling validate/revalidate/repaint on the component directly but that didn't help.

*Output formats*

The output formats combobox now reflects compiler support. However, there's the same problem as with the bibliographyPanel that I don't know how to change them immediately. So now if you switch compiler, changes will only be visible if you close and reopen the run config editor.

Example:
* Select Tectonic as compiler
* Close and reopen the run config window
* Select html as output format and a browser as pdf viewer (okay the html output doesn't make any sense, but that's a different problem)
* Change back to pdflatex
* Close and reopen the window
* Format is back to pdf
